### PR TITLE
Catch edge-case in train.py when the model may be overfit

### PR DESCRIPTION
**Pull request by dscripka**
Fixes a bug where metric calculation will raise an exception of a prediction Tensor is empty. This may happen when the model becomes overfit to the training data.

### CODE REVIEW

**Review by tauras129 — APPROVED** [review]
looks good enough for me
